### ISSUE TITLE
TODO for release notes 0.14.0

### ISSUE THREAD

**Issue by laanwj** [bitcoin/bitcoin#8455]
Post things here that need explicit mention in 0.14 release notes

I'd strongly prefer for pulls to just update `doc/release-notes.md` directly, but if they somehow slip by, mention them here.

Overview:
- [x] ZMQ on Windows (#8238) (done in wiki)
- Changed RPC calls:
  - [x] `getaddednodeinfo` got rid of first argument (#8272) (done in wiki)
  - [x] `getinfo` was deprecated (but not yet removed) (#8780) (done in wiki)
  - [x] `preciousblock` RPC call #6996
  - [x] `importmulti` RPC: #7551 
  - [x] `getmemoryinfo` #8753
  - [x] `setnetworkactive RPC (#8996)
- [x] Mention that the GUI confirmation target has been changed from the default of 25 to 6 (#9036).
- [x] Mempool dump to disk (#8448) (done in wiki)
- Several memory usage improvements 
  - [x] (shared pointers, LockedPool #8753)
  - [ ] Share unused mempool space with coin cache (#8610) 
- [x] Qt RPC nested commands: https://github.com/bitcoin/bitcoin/pull/7783 (done in wiki)
- [x] Qt sensitive data filter: https://github.com/bitcoin/bitcoin/pull/8877 (done in wiki)
- [x] Removal of priority estimation (#7730).
- [ ] Address relay changes (#8594, #8661, #8740).
- [x] Named RPC arguments (#8811).
- [x] Manual block file pruning (#7871)
- [x] Deprecate use of getrawtransaction for confirmed transactions with unspent outputs without -txindex (#9520).

~~**Note:** we're doing an experiment with editing the release notes draft in a github wiki.
The most up to date version of the release notes is thus at https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.14.0-Release-notes . Feel free to edit them there if you want to add any of these items.~~

**Comment by fanquake** [comment]
ZMQ on Windows from #8238

**Comment by MarcoFalke** [comment]
*  #8699: Remove createwitnessaddress RPC command

**Comment by MarcoFalke** [comment]
Is the release schedule for 0.14 out yet? (I couldn't find it anywhere)

**Comment by laanwj** [comment]
@MarcoFalke thanks for the reminder, see #8719

**Comment by laanwj** [comment]
[rpc] Deprecate getinfo #8780

**Comment by laanwj** [comment]
As for deprecating getinfo, the following table may be useful for in the release notes

`getinfo` field            | Moved to
-------------------------- | --------------------------------------
`"version"`                | `getnetworkinfo()["version"]`
`"protocolversion"`        | `getnetworkinfo()["protocolversion"]`
`"walletversion"`          | `getwalletinfo()["walletversion"]`
`"balance"`                | `getwalletinfo()["balance"]`
`"blocks"`                 | `getblockchaininfo()["blocks"]`
`"timeoffset"`             | `getnetworkinfo()["timeoffset"]`
`"connections"`            | `getnetworkinfo()["connections"]`
`"proxy"`                  | `getnetworkinfo()["networks"][0]["proxy"]`
`"difficulty"`             | `getblockchaininfo()["difficulty"]`
`"testnet"`                | `getnetworkinfo()["chain"] == "test"`
`"keypoololdest"`          | `getwalletinfo()["keypoololdest"]`
`"keypoolsize"`            | `getwalletinfo()["keypoolsize"]`
`"unlocked_until"`         | `getwalletinfo()["unlocked_until"]`
`"paytxfee"`               | `getwalletinfo()["paytxfee"]`
`"relayfee"`               | `getnetworkinfo()["relayfee"]`
`"errors"`                 | `getnetworkinfo()["errors"]`

**Comment by laanwj** [comment]
> #8699: Remove createwitnessaddress RPC command

This is one for the 0.13.1 release notes, as it was backported in #8815.

**Comment by jonasschnelli** [comment]
`preciousblock` RPC call

**Comment by laanwj** [comment]
`importmulti` RPC: #7551

**Comment by jonasschnelli** [comment]
Mention that the GUI confirmation target has been changed from the default of 25 to 2 #8989 (or maybe 6 after #9036)

**Comment by laanwj** [comment]
Yes, also the RPC default will go to 6 if #9036.

New RPC: `getmemoryinfo` #8753

**Comment by MarcoFalke** [comment]
Created a label "Needs release notes" for pulls to be tagged with. Could be useful?

**Comment by laanwj** [comment]
@MarcoFalke yes good idea

**Comment by sipa** [comment]
GUI now has a modal out of sync layer (#8371).

**Comment by sipa** [comment]
Share unused mempool space with coin cache (#8610).

**Comment by sipa** [comment]
Mempool dump to disk (#8448).

**Comment by sipa** [comment]
Change default confirm target from 2 to 6 (#9036).

**Comment by sipa** [comment]
Several memory usage improvements (shared pointers, LockedPool #8753)

**Comment by sipa** [comment]
togglenetwork RPC (#8996).

**Comment by sipa** [comment]
Update to LevelDB 1.19 (#8613).

**Comment by sdaftuar** [comment]
estimatefee for a target of 1 block has been disabled (this part was merged into 0.13.2 as well), and the gui slider will stop at 2 now (new for 0.14) (#9239)

**Comment by jonasschnelli** [comment]
Qt RPC nested commands: https://github.com/bitcoin/bitcoin/pull/7783

**Comment by jonasschnelli** [comment]
Qt sensitive data filter: https://github.com/bitcoin/bitcoin/pull/8877

**Comment by sipa** [comment]
Return txid even if ATMP fails for new transaction (#9302).

**Comment by sipa** [comment]
Removal of priority estimation (#7730).

**Comment by sipa** [comment]
Address relay changes (#8584, #8661, #8740).

**Comment by sipa** [comment]
Named RPC arguments (#8811).

**Comment by luke-jr** [comment]
#9263 needs to be merged in

**Comment by laanwj** [comment]
> Update to LevelDB 1.19 (#8613).

Do you have a specific reason this needs special mention in the release notes? (besides a normal changelog line) Like other dependency upgrades, it's not a user-facing change.

**Comment by MarcoFalke** [comment]
> Return txid even if ATMP fails for new transaction (#9302).
> Prefer coins that have fewer ancestors (#9262).

Those are already in 0.13 so don't need to be put in the notes for 0.14?

**Comment by laanwj** [comment]
> Those are already in 0.13 so don't need to be put in the notes for 0.14?

Indeed, good catch. Release notes are a diff to the last minor release on previous major. I guess an exception could be made if it was somehow forgotten there.

**Comment by sipa** [comment]
> > Update to LevelDB 1.19 (#8613).
>
> Do you have a specific reason this needs special mention in the release notes? (besides a normal changelog line) Like other dependency upgrades, it's not a user-facing change.

Fair enough.

**Comment by laanwj** [comment]
#7871: Manual block file pruning

**Comment by fanquake** [comment]
I've updated to top post to include everything added here so far.

**Comment by sipa** [comment]
#9520 Deprecate use of getrawtransaction for confirmed transactions with unspent outputs without -txindex.

**Comment by laanwj** [comment]
Note: we're doing an experiment with editing the release notes draft in a github wiki.
The most up to date version of the release notes is thus at https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.14.0-Release-notes . Feel free to edit them there if you want to add any of these items.

**Comment by MarcoFalke** [comment]
#8456 bumpfee

**Comment by kallewoof** [comment]
Maybe mentioning changes to `listsinceblock` as these now use fork point rather than the relative height of the reorged block vs current main chain (#9516).

I'm also confused by the wording "This is a new major version release", since traditionally, the version is MAJOR.MINOR.PATCH, and this is thus a new MINOR version release. Perhaps "major" is referring to a major change in codebase only?

**Comment by laanwj** [comment]
> since traditionally, the version is MAJOR.MINOR.PATCH, and this is thus a new MINOR version release

Traditionally for other projects perhaps. In bitcoin core we've always used "major" for `0.x.0` and "minor" for `0.x.y`. To be honest I don't think that's going to ever change, unless for some freak reason the first number ever gets bumped, but that's a completely different discussion that doesn't belong here.

**Comment by MarcoFalke** [comment]
We might just drop the `0.` prefix if it is always present and thus, count releases like firefox and similar. But agree it is off topic here.

**Comment by MarcoFalke** [comment]
Could make sense to sync the release notes from the wiki to our repo in a pull request, so we can see if all looks fine.

**Comment by morcos** [comment]
Need to add explanation of at least some of the new minimum fees (incrementalRelayFee, dustRelayFee and blockMinTxFee) as well as the fact that it is now unnecessary to set minRelayTxFee

**Comment by MarcoFalke** [comment]
bumpfee is still missing?

**Comment by laanwj** [comment]
#9878 added mention of bumpfee (thanks @ryanofsky), I think this is done, closing.
